pkg/topology/mock/mock.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mock
+
+import (
+	"context"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+)
+
+type TopologyDriver struct {
+	peers      []swarm.Address
+	addPeerErr error
+}
+
+func NewTopologyDriver() *TopologyDriver {
+	return &TopologyDriver{}
+}
+
+func (d *TopologyDriver) SetAddPeerErr(err error) {
+	d.addPeerErr = err
+}
+
+func (d *TopologyDriver) AddPeer(_ context.Context, addr swarm.Address) error {
+	if d.addPeerErr != nil {
+		return d.addPeerErr
+	}
+
+	d.peers = append(d.peers, addr)
```

Comment:
i wonder if we need a lock here

Reply:
I did not expect to use it concurrently now, but I can add it if you think it might be useful for the future as well?

Reply:
well, if we are mutating a slice then i think it might be a good idea. otherwise we would have to enforce that any test that uses the mock cannot introduce any concurrency at all, which would be difficult on the long run

Reply:
Done